### PR TITLE
Lock acquisition with timeout implementation.

### PR DESCRIPTION
**Pull request by jerome-benoit**
@OliveGerste have checked that's what he's expecting

Unit tests will be added in a later PR. 

Signed-off-by: Jérôme Benoit <jerome.benoit@sap.com>

### CODE REVIEW

File: src/locking/LockingManager.ts [hunk 1]
```diff
@@ -55,11 +55,62 @@ export default class LockingManager {
...
+          while (!timeoutReached) {
+            try {
+              await LockingStorage.insertLock(lock);
+              break;
+            } catch {
+              await Utils.sleep(1000);
+            }
+          }
+          if (timeoutReached) {
+            throw Error(`Lock acquisition timeout ${timeout}ms reached`);
```
Comment (OliveGerste) [inline src/locking/LockingManager.ts:76]: Should we also log something here?

**Review by OliveGerste — COMMENTED** [review]
I tested it in my local environment and it worked. 
To see if it is working in real life, we could use it in the smart charging integration and start two transactions at the exact same time. 
If smart charging runs twice, it works.

File: src/locking/LockingManager.ts (hunk 1 above)
Comment (jerome-benoit) [reply src/locking/LockingManager.ts:76]: The catch should log error thrown.

**Comment by jerome-benoit** [comment]
> I tested it in my local environment and it worked.
> To see if it is working in real life, we could use it in the smart charging integration and start two transactions at the exact same time.
> If smart charging runs twice, it works.

I will try to find time to add some unit tests, which might be a bit tricky to implement, unless you need it urgently and then I create another issue to add the unit test later on.  

I'm not very happy with the busy loop wait and retry (CPU hog) but I think it should be enough. The other idea I have was to wait and retry by looping over an array of promises trying to get the lock.

**Review by LucasBrazi06 — APPROVED** [review]
Gut!

**Review by LucasBrazi06 — APPROVED** [review]

**Pull request merged**